wpilibj/src/main/java/edu/wpi/first/wpilibj/drive/HDrive.java

```diff
@@ -0,0 +1,277 @@
+
+  /**
+   * Construct a HDrive.
+   *
+   * <p>
+   * To pass multiple motors per side, use a {@link
+   * edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup}. If a motor needs to
+   * be inverted, do
+   * so before passing it in.
+   *
+   * @param leftMotor  Left motor.
+   * @param rightMotor Right motor.
+   * @param lateralMotor Lateral motor.
+   *
+   * <p>Inputs smaller then {@value edu.wpi.first.wpilibj.drive.RobotDriveBase#kDefaultDeadband} will
+   * be set to 0, and larger values will be scaled so that the full range is still used. This deadband
+   * value can be changed with {@link #setDeadband}.
+   *
+   * <p>{@link edu.wpi.first.wpilibj.MotorSafety} is enabled by default. The tankDrive or arcadeDrive,
+   * methods should be called periodically to avoid Motor Safety timeouts.
+   */
+  public HDrive(MotorController leftMotor, MotorController rightMotor, MotorController lateralMotor) {
+    requireNonNullParam(leftMotor, "leftMotor", "HDrive");
+    requireNonNullParam(rightMotor, "rightMotor", "HDrive");
+    requireNonNullParam(lateralMotor, "lateralMotor", "HDrive");
+
+    m_leftMotor = leftMotor;
+    m_rightMotor = rightMotor;
+    m_lateralMotor = lateralMotor;
+    SendableRegistry.addChild(this, m_leftMotor);
+    SendableRegistry.addChild(this, m_rightMotor);
+    SendableRegistry.addChild(this, m_lateralMotor);
+    instances++;
+    SendableRegistry.addLW(this, "HDrive", instances);
+  }
+
+  @Override
+  public void close() {
+    SendableRegistry.remove(this);
+  }
+
+  /**
+   * Arcade drive method for differential drive platform. The calculated values
+   * will be squared to
+   * decrease sensitivity at low speeds.
+   *
+   * @param xSpeed    The robot's speed along the X axis [-1.0..1.0]. Forward is
+   *                  positive.
+   * @param zRotation The robot's rotation rate around the Z axis [-1.0..1.0].
+   *                  Counterclockwise is
+   *                  positive.
+   * @param ySpeed The robot's speed along the robots lateral axis [-1.0..1.0]. Right is positive.
+   */
+  public void arcadeDrive(double xSpeed, double zRotation, double ySpeed) {
+    arcadeDrive(xSpeed, zRotation, ySpeed, true);
+  }
+
+  /**
+   * Arcade drive method for differential drive platform.
+   *
+   * @param xSpeed       The robot's speed along the X axis [-1.0..1.0]. Forward
+   *                     is positive.
+   * @param zRotation    The robot's rotation rate around the Z axis [-1.0..1.0].
+   *                     Counterclockwise is
+   *                     positive.
+   * @param ySpeed The robot's speed along the robots lateral axis [-1.0..1.0]. Right is positive.
+   * @param squareInputs If set, decreases the input sensitivity at low speeds.
+   */
+  public void arcadeDrive(double xSpeed, double zRotation, double ySpeed, boolean squareInputs) {
+    if (!m_reported) {
+      HAL.report(
+          tResourceType.kResourceType_RobotDrive, tInstances.kRobotDrive2_DifferentialArcade, 2);
```

Comment:
We probably want another instance number assigned for this.

Reply:
If you tell me what number to implement I'll swap it over but I don't want to risk accidentally interfering with anything until a number for tResourceType.kResourceType_RobotDrive_HDriveArcade & tResourceType.kResourceType_RobotDrive_HDriveTank are chosen.